pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -675,14 +675,14 @@ func (m *kubeGenericRuntimeManager) AttachContainer(id kubecontainer.ContainerID
 
 // GetContainerLogs returns logs of a specific container.
 func (m *kubeGenericRuntimeManager) GetContainerLogs(pod *api.Pod, containerID kubecontainer.ContainerID, logOptions *api.PodLogOptions, stdout, stderr io.Writer) (err error) {
-	// Get logs directly from docker for in-process docker integration for
-	// now to unblock other tests.
-	// TODO: remove this hack after setting down on how to implement log
-	// retrieval/management.
-	if ds, ok := m.runtimeService.(dockershim.DockerLegacyService); ok {
-		return ds.GetContainerLogs(pod, containerID, logOptions, stdout, stderr)
+	status, err := m.runtimeService.ContainerStatus(containerID.ID)
```

Comment:
I don't think this is more reliable because you still rely on kubelet to pick a container ID to begin with. 
I am ok with this for now if we add a TODO comment for consolidating the code.